keycloak/sms-provider/src/main/java/org/sunbird/keycloak/utils/SunbirdModelUtils.java

```diff
@@ -37,29 +38,31 @@ public static UserModel getUserByNameEmailOrPhone(AuthenticationFlowContext cont
     KeycloakSession session = context.getSession();
     logger.info("SunbirdModelUtils@getUser " + username);
     if (username.matches(numberRegex)) {
-      List<UserModel> userModels = session.users().searchForUserByUserAttribute(
-          KeycloakSmsAuthenticatorConstants.ATTR_MOBILE, username, context.getRealm());
-      if (userModels != null && !userModels.isEmpty()) {
+        Stream<UserModel> userModels = session.users().searchForUserByUserAttributeStream(
+              context.getRealm(), KeycloakSmsAuthenticatorConstants.ATTR_MOBILE, username);
+        //try the whole thing with streams if possible
+        List<UserModel> userModelsList = userModels.collect(Collectors.toList());
+      if (userModelsList != null && !userModelsList.isEmpty()) {
         // multiple user found for same attribute
-    	for(UserModel model : userModels) {
+    	for(UserModel model : userModelsList) {
```

Comment:
I think the variable name could have left as is : userModels
Due to this naming change, it is bringing in insignificant changes to this list.

Reply:
Yes agreed.